CompilerSource/backend/GameData.cpp

```diff
@@ -445,33 +527,64 @@ GameData::GameData(EnigmaStruct *es): filename(es->filename ?: "") {
   cout << "Transfer complete." << endl << endl;
 }
 
-GameData::GameData(const buffers::Project &proj): filename("") {
-  cout << "Flattening tree." << endl;
+GameData::GameData(const buffers::Project &proj): filename("") {}
 
-  FlattenTree(proj.game().root());
-
-  cout << "Transfer complete." << endl << endl;
-}
-
-void GameData::FlattenTree(const buffers::TreeNode &root) {
+int FlattenTree(const buffers::TreeNode &root, GameData *gameData) {
+  int error = 0;
   switch (root.type_case()) {
     case TypeCase::kFolder: break;
-    case TypeCase::kSprite: /*sprites.emplace_back(root.sprite(), root.name());*/ break;
-    case TypeCase::kSound: /*sounds.emplace_back(root.sound(), root.name());*/ break;
-    case TypeCase::kBackground: /*backgrounds.emplace_back(root.background(), root.name());*/ break;
-    case TypeCase::kPath: paths.emplace_back(root.path(), root.name()); break;
-    case TypeCase::kScript: scripts.emplace_back(root.script(), root.name()); break;
-    case TypeCase::kShader: shaders.emplace_back(root.shader(), root.name()); break;
-    case TypeCase::kFont: fonts.emplace_back(root.font(), root.name()); break;
-    case TypeCase::kTimeline: timelines.emplace_back(root.timeline(), root.name()); break;
-    case TypeCase::kObject: objects.emplace_back(root.object(), root.name()); break;
-    case TypeCase::kRoom: rooms.emplace_back(root.room(), root.name()); break;
-    case TypeCase::kInclude: /*includes.emplace_back(root.include());*/ break;
-    case TypeCase::kSettings: /*settings.emplace_back(root.settings());*/ break;
-    default: cout << "- Not transferring " << root.name() << endl; break;
+    case TypeCase::kSprite: {
+      std::vector<ImageData> subimages;
+      const buffers::resources::Sprite &sprite = root.sprite();
+      for (auto sub : sprite.subimages()) {
+        ImageData data = loadImageData(sub, error);
+        if (error) return -1; // sprite load error
+        subimages.emplace_back(data);
+      }
+      gameData->sprites.emplace_back(sprite, root.name(), subimages);
+      break;
+    }
+    case TypeCase::kSound: {
+      BinaryData data = loadBinaryData(root.sound().data(), error);
+      if (error) return -2; // sound load error
+      gameData->sounds.emplace_back(root.sound(), root.name(), data);
+      break;
+    }
+    case TypeCase::kBackground: {
+      ImageData data = loadImageData(root.background().image(), error);
+      if (error) return -3; // background load error
+      gameData->backgrounds.emplace_back(root.background(), root.name(), data);
+      break;
+    }
+    case TypeCase::kPath: gameData->paths.emplace_back(root.path(), root.name()); break;
```

Comment:
The lack of alignment here perturbs me greatly.

Reply:
Ok: b9f521ee276c9f99742e2286edf638ef2575f7d5

I think that's the alignment you want? I don't know.